tutorials/intellijDebugger.md

```diff
@@ -6,7 +6,140 @@ pageNav: 3
 
 # {{ title }}
 
+Before diving into tracing code with IntelliJ IDEA’s debugger, it’s essential to understand key debugger features and their purpose.
+
 This video (from LaunchCode) gives a pretty good explanation of how to use the IntelliJ IDEA debugger.
 
 @[youtube](1bCgzjatcr4)
 
+## Starting A Debugging Session
+
+To utilise the debugger, we need to make sure that the code is running in debug mode.
+
+### Run Debug Mode
+
+<box type="info" seamless>
+
+**Video reference:** [3.53 - 4.00](https://youtu.be/1bCgzjatcr4?si=fIEvUVhXHwKQeOlc&t=233): Run your program in debug mode to use debugging features.
+</box>
+
+Go to `Run` > `Debug 'Main'` or click on the bug icon in the top-right corner of the IntelliJ IDEA window.
+
+### Setting Breakpoints
+
+<box type="info" seamless>
+
+**Video reference:** [3.09 - 3.30](https://youtu.be/1bCgzjatcr4?si=KkJw2Cri-ERmqgAJ&t=189): Set a breakpoint by clicking on the line number in the left gutter like this!
+</box>
+
+To set a breakpoint in IntelliJ IDEA:
+
+- Click on the line number in the left gutter.
+
+Breakpoints are useful for:
+
+- Pausing execution at method entry points
+- Examining the program state before complex operations
```

Comment:
For each segment, we can also give a cropped screenshot, to show the feature.

Reply:
I have updated the PR with the video embedded. With that change, would adding a screenshot make each section too convoluted?

Reply:
> I have updated the PR with the video embedded. With that change, would adding a screenshot make each section too convoluted?

@ruishanteo Screenshot takes only a second to grasp (especially good for those who watched the video before and not revisiting to recall something). Those who still don't get it from the screenshot will go to spend more time watching the video.